docs/tutorial/building-a-dataset/index.md

```diff
@@ -0,0 +1,145 @@
+
+index_date = "2024-03-31"
+
+diabetes_codes = codelist_from_csv("codelists/dm_cod.csv", column="code")
+resolved_codes = codelist_from_csv("codelists/dmres_cod.csv", column="code")
+proteinuria_codes = codelist_from_csv("codelists/prt_cod.csv", column="code")
+microalbuminuria_codes = codelist_from_csv("codelists/mal_cod.csv", column="code")
+ace_codes = codelist_from_csv("codelists/ace_cod.csv", column="code")
+arb_codes = codelist_from_csv("codelists/arb_cod.csv", column="code")
+
+aged_17_or_older = (index_date - patients.date_of_birth).years >= 17
+was_alive = patients.date_of_death.is_null() | (patients.date_of_death < index_date)
+was_registered = (
+    practice_registrations.where(practice_registrations.start_date < index_date)
+    .except_where(practice_registrations.end_date < index_date)
+    .exists_for_patient()
+)
+
+last_diagnosis_date = (
+    clinical_events.where(clinical_events.snomedct_code.is_in(diabetes_codes))
+    .sort_by("date")
+    .last_for_patient()
+    .date
+)
+last_resolved_date = (
+    clinical_events.where(clinical_events.snomedct_code.is_in(resolved_codes))
+    .sort_by("date")
+    .last_for_patient()
+    .date
+)
+
+has_unresolved_diabetes = last_diagnosis_date.is_not_null() & (
+    last_resolved_date.is_null() | (last_resolved_date < last_diagnosis_date)
+)
+
+on_register = aged_17_or_older & was_alive & was_registered & has_undiagnosed_diabetes
+
+has_proteinuria_diagnosis = (
+    clinical_events.where(clinical_events.date.is_before(index_date))
+    .where(clinical_events.snomedct_code.is_in(proteinuria_codes))
+    .exists_for_patient()
+)
+
+has_microalbuminuria_diagnosis = (
+    clinical_events.where(clinical_events.date.is_before(index_date))
+    .where(clinical_events.snomedct_code.is_in(microalbuminuria_codes))
+    .exists_for_patient()
+)
+
+has_arb_treatment = (
+    medications.where(medications.date.is_between(index_date - days(180), index_date))
+    .where(medications.dmd_code.is_in(arb_codes))
+    .exists_for_patient()
+)
+
+has_ace_treatment = (
+    medications.where(medications.date.is_between(index_date - days(180), index_date))
+    .where(medications.dmd_code.is_in(ace_codes))
+    .exists_for_patient()
+)
+
+```
+
+Note that it is essential that the dataset you create is given the name `dataset`.
+
+> Question: what happens if you try to set the dataset's population to something that is not a boolean patient series?
+>
+> Question: what happens if you try to set a dataset column to something that is not a patient series?
+>
+> Question: what happens if you try to reuse a name for a dataset's column?
+
+
+
+[1]: https://www.opencodelists.org/codelist/nhsd-primary-care-domain-refsets/prt_cod/
+[2]: https://www.opencodelists.org/codelist/nhsd-primary-care-domain-refsets/mal_cod/
+[3]: https://www.opencodelists.org/codelist/nhsd-primary-care-domain-refsets/ace_cod/
+[4]: https://www.opencodelists.org/codelist/nhsd-primary-care-domain-refsets/arb_cod/
```

Comment:
two broken links

Reply:
not sure which are correct
https://www.opencodelists.org/codelist/nhsd-primary-care-domain-refsets/?q=arb
https://www.opencodelists.org/codelist/nhsd-primary-care-domain-refsets/?q=ace

Reply:
So I'd hoped to be able to upload these codelists but they're not published by QOF unlike the others.  So we're going to use https://www.opencodelists.org/codelist/opensafely/angiotensin-ii-receptor-blockers-arbs/ and https://www.opencodelists.org/codelist/opensafely/ace-inhibitor-medications/.